tests/functional/Contract.php

```diff
@@ -99,4 +99,79 @@ public function testClone()
             )->isIdenticalTo(1, 'Missing relation with ' . $rel_class);
         }
     }
+
+    public function getSpecificValueToDisplayProvider()
+    {
+        $duration = 6;
+        $past_date = '2020-01-01';
+        $futur_date = date('Y-01-01', strtotime('+1 year'));
+        $futur_date_end = date('Y-m-d', strtotime("+$duration months", strtotime($futur_date)));
+        return [
+            [
+                'field' => '_virtual_expiration',
+                'values' => [
+                    'begin_date' => $past_date,
+                    'duration' => $duration,
+                    'renewal' => \Contract::RENEWAL_NEVER
+                ],
+                'expected' => "<span class='red'>2020-07-01</span>"
+            ],
+            [
+                'field' => '_virtual_expiration',
+                'values' => [
+                    'begin_date' => $past_date,
+                    'duration' => $duration,
+                    'renewal' => \Contract::RENEWAL_TACIT
+                ],
+                'expected' => "2024-07-01"
```

Comment:
This test will fail once this date will be passed. I guess the `Infocom::getWarrantyExpir()` must be changed to use the `$_SESSION['glpi_currenttime']` value, in order to be able to force it in this test case.